web/src/main/java/org/akaza/openclinica/service/UserServiceImpl.java

```diff
@@ -387,6 +388,87 @@ public void extractParticipantsInfo(String studyOid, String siteOid, String acce
+        Study study = studyDao.findByOcOID(studyOid);
+
+
+        // Get all list of StudySubjects by studyId
+        StudySubject studySubject = studySubjectDao.findByLabelAndStudyOrParentStudy(participantID, study);
+        OCUserDTO ocuserDTO = null;
+        StudyParticipantDetailDTO spDTO= new StudyParticipantDetailDTO();
+        
+        if(studySubject == null) {
+        	String errorCode ="errorCode.participantNotAvailable";
+        	String msg = "Can't find the participant with ID:" + participantID;
+        	throw new OpenClinicaSystemException(errorCode, msg);
+        }
+            
+        try {
+          
+            	if (!studySubject.getStatus().equals(Status.DELETED)
+                        && !studySubject.getStatus().equals(Status.AUTO_DELETED)) {
+
+            		 /**
+                     * OC-10640
+                     * AC4: Participant contact information and their Participate related information should only be returned
+                     *  for participants that are in available or signed status.
+                     */
+                    if (studySubject.getStatus().equals(Status.AVAILABLE)
+                            || studySubject.getStatus().equals(Status.SIGNED)) {
+
+                        
+                        //Get accessToken from Keycloak                      
+                    	ocuserDTO = buildOcUserDTO(studySubject,incRelatedInfo);
+                        ParticipantAccessDTO participantAccessDTO = getAccessInfo(accessToken, siteOid, studySubject.getLabel(), customerUuid, userAccountBean,incRelatedInfo,incRelatedInfo);
+                        
+                        
+                        if (participantAccessDTO != null && participantAccessDTO.getAccessCode() != null && incRelatedInfo) {
+                        	ocuserDTO.setAccessCode(participantAccessDTO.getAccessCode());
+                        }                                             
+                    }
+                    
+                  
+		        	
+		        	spDTO.setSubjectOid(studySubject.getOcOid());
+		        	spDTO.setSubjectKey(studySubject.getLabel());
+		        	
+		        	if(incRelatedInfo) {
+		        		spDTO.setStatus(studySubject.getStatus().getName());
+		        		spDTO.setAccessCode(ocuserDTO.getAccessCode());
+		        	}
+		        	
+		        	
+		        	
+		        	if(ocuserDTO != null) {
+		        		spDTO.setFirstName(ocuserDTO.getFirstName());
+		        		spDTO.setLastName(ocuserDTO.getLastName());
+		        		spDTO.setEmail(ocuserDTO.getEmail());
+		        		spDTO.setMobileNumber(ocuserDTO.getPhoneNumber());
+		        		
+		        	}
+			        	
+			   
+			        
+			        return spDTO;
+                    
+            	}            		
+               
+            
+           
+        } catch (Exception e) {
+           
+            logger.error(" Access code Job Creation Failed :"+ e.getMessage());
```

Comment:
instead of concatenating error message, you can pass exception as an argument - log.error("message", e);

Reply:
done